perpVects/QFPHelpers.cpp

```diff
@@ -37,19 +37,51 @@ std::ostream& operator<<(std::ostream& os, const unsigned __int128 &i){
   return os;
 }
 
-std::string
-getSortName(sort_t val){
-  switch(val){
-  case lt:
-    return "lt";
-  case gt:
-    return "gt";
-  case bi:
-    return "bi";
-  case def:
-    return "us";
-  default:
-    return "something bad happened, undefined sort type";
+namespace FPHelpers{
+    auto
+  swap_float_int(unsigned int val) -> float{
```

Comment:
Why the use of auto when you know the return type?  The only reason for that kind of notation is to be able to use the parameters in a decltype() call to determine the return type.

I suggest just using for example:

float swap_float_int(uint32_t val) { ... }